core/trino-main/src/test/java/io/trino/connector/MockConnectorFactory.java

```diff
@@ -428,42 +429,52 @@ public Builder withData(Function<SchemaTableName, List<List<?>>> data)
 
         public Builder withProcedures(Set<Procedure> procedures)
         {
+            provideAccessControl = true;
             this.procedures = procedures;
             return this;
         }
 
         public Builder withListRoleGrants(ListRoleGrants roleGrants)
         {
+            provideAccessControl = true;
             this.roleGrants = requireNonNull(roleGrants, "roleGrants is null");
             return this;
         }
 
         public Builder withSchemaGrants(Grants<String> schemaGrants)
         {
+            provideAccessControl = true;
             this.schemaGrants = schemaGrants;
             return this;
         }
 
         public Builder withTableGrants(Grants<SchemaTableName> tableGrants)
         {
+            provideAccessControl = true;
             this.tableGrants = tableGrants;
             return this;
         }
 
         public Builder withRowFilter(Function<SchemaTableName, ViewExpression> rowFilter)
         {
+            provideAccessControl = true;
             this.rowFilter = rowFilter;
             return this;
         }
 
         public Builder withColumnMask(BiFunction<SchemaTableName, String, ViewExpression> columnMask)
         {
+            provideAccessControl = true;
             this.columnMask = columnMask;
             return this;
         }
 
         public MockConnectorFactory build()
         {
+            Optional<MockConnectorAccessControl> accessControl = Optional.empty();
+            if (provideAccessControl) {
+                accessControl = Optional.of(new MockConnectorAccessControl(schemaGrants, tableGrants, rowFilter, columnMask));
+            }
```

Comment:
Again, why do we need this? Is it because `MockConnectorAccessControl`'s initialization overhead?

Reply:
Thanks!